### ISSUE TITLE
cmake issue with v2.2.2

### ISSUE THREAD

**Issue by davidbrochart** [BlueBrain/HighFive#394]
When updating `highfive` from v2.1.1 to v2.2.2 in xtensor-io in https://github.com/xtensor-stack/xtensor-io/pull/124, we get the following error (see [the logs](https://dev.azure.com/xtensor-stack/xtensor-stack/_build/results?buildId=1562&view=logs&j=6f6950ed-118e-5110-b7d1-309d986feacb&t=203e8e97-b2de-59ca-3bb6-c6ccc39f33a4&l=189)):
```
CMake Error: install(EXPORT "xtensor-io-targets" ...) includes target "xtensor-io" which requires target "libdeps" that is not in any export set.
```
A work-around was to [add libdeps to xtensor-io's target](https://github.com/xtensor-stack/xtensor-io/pull/124/commits/279ffabc93baf638a7e041c19c8f7e8939ed4681), but is there a way to fix it upstream?

**Comment by tdegeus** [comment]
See https://github.com/BlueBrain/HighFive/pull/257 for my earlier initiative

**Comment by SylvainCorlay** [comment]
Thanks @tdegeus. Beyond your (very welcome) modernization of Highfive's cmake, it seems that there is a legitimate bug here.

Highfive's cmake should probably install the libdeps target.

**Comment by kerim371** [comment]
Concerning CMake one additional note:

Now HighFive tries to find HDF5 with command `find_package(HDF5 REQUIRED)` which (as far as I know) involves CMake module to find `HDF5`.

On windows it is always difficult to find hdf5 this way: all libraries compiled by me not added to any env::var. 
When I try to configure HighFive with CMake GUI and after I manually set all path to hdf5 I get error as shown in the picture.

So I propose to use `find_package(HDF5 REQUIRED CONFIG)` and that would allow to simply set `HDF5_DIR` to `hdf5-config.cmake` and `HDF5` is found.

![image](https://user-images.githubusercontent.com/43808863/102512626-4f54d400-409b-11eb-9cae-318fe68de179.png)

**Comment by SylvainCorlay** [comment]
It seems this error is still there with Hifive v2.3.1. We are still pinning on v2.1 because of it...

```
CMake Error in CMakeLists.txt:
  export called with target "xtensor-io" which requires target "libdeps" that
  is not in any export set.
```

**Comment by 1uc** [comment]
We've rewritten the HighFive CMake related code for `v3` and are close to of have merge the changes. I'm closing this issue as outdated. Please open a new Issue if the problem persists after #897 got merged.
